Gordon360/Controllers/HousingController.cs

```diff
@@ -233,7 +233,7 @@ public ActionResult<ApartmentApplicationViewModel[]> GetAllApartmentApplication(
             var authenticatedUserIdString = User.FindFirst(ClaimTypes.NameIdentifier).Value;
 
             
-            ApartmentApplicationViewModel[] result = _housingService.GetAllApartmentApplication();
+            ApartmentApplicationViewModel[] result = _housingService.GetAllApartmentApplication() ?? new ApartmentApplicationViewModel[0];
```

Comment:
I'm confused, this doesn't look like the same changes as the original commit at all.

Reply:
I think if you want to recover the original changes, you need to extract `applicationDetails.ApartmentChoices` on lines 133 and 162 into a temp variable that has a default value of an empty array.

Alternatively, just ensure that the `IHousingService` methods `EditApplication` and `SaveApplication` accept a nullable array of apartment choices and handle it appropriately.

Reply:
The code has been changed so much that the original issues do not exist. I took the intention of the previous commit and fixed the one instance where it was still an issue.

Reply:
Ok, I see.

I don't think there's any reason to merge this commit though. You're actually degrading the route by adding a default empty array, because it breaks the null check directly below this. No we're returning an empty array instead of not found (which I prefer on principle, but they probably expect it to return not found in that circumstance.)